components/brave_wallet/browser/solana_transaction.cc

```diff
@@ -274,16 +275,24 @@ SolanaTransaction::GetSignedTransactionBytes(
   // for this transaction.
   uint8_t num_of_sig = 0;
   for (const auto& signer : signers) {
-    if (*selected_account == signer) {
+    if (base::EqualsCaseInsensitiveASCII(selected_account->address, signer)) {
       if (selected_account_signature) {
         transaction_bytes.insert(transaction_bytes.end(),
                                  selected_account_signature->begin(),
                                  selected_account_signature->end());
       } else {
-        std::vector<uint8_t> signature =
-            keyring_service->SignMessageBySolanaKeyring(signer, message_bytes);
-        transaction_bytes.insert(transaction_bytes.end(), signature.begin(),
-                                 signature.end());
+        for (auto& acc : all_accounts->accounts) {
+          // TODO(apymyshev): how permissions should be checked? What if account
+          // is missing?
+          if (base::EqualsCaseInsensitiveASCII(acc->address, signer)) {
+            std::vector<uint8_t> signature =
+                keyring_service->SignMessageBySolanaKeyring(*acc->account_id,
+                                                            message_bytes);
+            transaction_bytes.insert(transaction_bytes.end(), signature.begin(),
+                                     signature.end());
+            break;
+          }
+        }
```

Comment:
Not sure I understand what is going on here

Reply:
https://github.com/brave/brave-core/pull/18930#discussion_r1240558009